latch_cli/services/cp/path_utils.py

```diff
@@ -0,0 +1,144 @@
+import re
+import urllib.parse
+from urllib.parse import urljoin, urlparse
+
+from latch_cli.services.cp.exceptions import PathResolutionError
+
+# todo(ayush): need a better way to check if "latch" has been appended to urllib
+if urllib.parse.uses_netloc[-1] != "latch":
+    urllib.parse.uses_netloc.append("latch")
+    urllib.parse.uses_relative.append("latch")
+
+
+latch_url_regex = re.compile(r"^(latch)?://")
+
+
+def is_remote_path(path: str) -> bool:
+    return latch_url_regex.match(path) is not None
+
+
+def urljoins(*args: str, dir: bool = False) -> str:
+    """Construct a URL by appending paths
+
+    Paths are always joined, with extra `/`s added if missing. Does not allow
+    overriding basenames as opposed to normal `urljoin`. Whether the final
+    path ends in a `/` is still significant and will be preserved in the output
+
+    >>> urljoin("latch:///directory/", "another_directory")
+    latch:///directory/another_directory
+    >>> # No slash means "another_directory" is treated as a filename
+    >>> urljoin(urljoin("latch:///directory/", "another_directory"), "file")
+    latch:///directory/file
+    >>> # Unintentionally overrode the filename
+    >>> urljoins("latch:///directory/", "another_directory", "file")
+    latch:///directory/another_directory/file
+    >>> # Joined paths as expected
+
+    Args:
+        args: Paths to join
+        dir: If true, ensure the output ends with a `/`
+    """
+
+    res = args[0]
+    for x in args[1:]:
+        if res[-1] != "/":
+            res = f"{res}/"
+        res = urljoin(res, x)
+
+    if dir and res[-1] != "/":
+        res = f"{res}/"
+
+    return res
+
+
+scheme = re.compile(
+    r"""
+    ^(
+        (latch://) |
+        (file://) |
+        (?P<implicit_url>://) |
+        (?P<absolute_path>/) |
+        (?P<relative_path>[^/])
+    )
+    """,
+    re.VERBOSE,
+)
+domain = re.compile(
+    r"""
+    ^(
+        | # empty
+        (?P<account_relative>
```

Comment:
we actually only care about `latch:///` and `latch://shared/` here
since those are the only ones dependent on current workspace